lib/vagrant-service-manager/binary_handlers/binary_handler.rb

```diff
@@ -146,6 +147,19 @@ def validate_url
 
       private
 
+      def set_proxy_environment
+        return if @machine.config.servicemanager.proxy.nil?
+
+        env_proxy = @machine.config.servicemanager.proxy
+        if @machine.config.servicemanager.proxy_user
+          user = @machine.config.servicemanager.proxy_user
+          password = @machine.config.servicemanager.proxy_password
+          env_proxy = env_proxy.sub(%r{(https?:\/\/)*}, "\\1#{user}:#{password}@")
+        end
+
+        ENV['https_proxy'] = env_proxy
```

Comment:
Again, a comment on why you set this variable and why you set it in this format.